signac/contrib/schema.py

```diff
@@ -136,13 +138,14 @@ def __repr__(self):
 
     def __contains__(self, key_or_keys):
         if isinstance(key_or_keys, str):
-            key_or_keys = key_or_keys.split('.')
-        return tuple(key_or_keys) in self._schema
+            return key_or_keys in self._schema
+        key_or_keys = '.'.join(key_or_keys)
+        return key_or_keys in self._schema
 
     def __getitem__(self, key_or_keys):
         if isinstance(key_or_keys, str):
-            key_or_keys = key_or_keys.split('.')
-        return self._schema.__getitem__(tuple(key_or_keys))
+            return self._schema.__getitem__(key_or_keys)
```

Comment:
This can be simplified. In general, you should never use `__getitem__` or other double underscore methods directly. You may sometimes need to use `getitem(self._schema, key_or_keys, default_value)` if you need to provide a default value if the key cannot be accessed, but if no defaults are needed, just use the `[]` accessor.
```suggestion
            return self._schema[key_or_keys]
```